lib/CustomFields/components/CustomFieldsForm/components/DeleteModal/DeleteModal.js

```diff
@@ -0,0 +1,116 @@
+import React, { Fragment, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  Modal,
+  ModalFooter,
+  Button,
+  Icon,
+  Layout,
+} from '@folio/stripes-components';
+
+import { FormattedMessage } from 'react-intl';
+import SafeHTMLMessage from '@folio/react-intl-safe-html';
+
+const propTypes = {
+  fetchUsageStatistics: PropTypes.func.isRequired,
+  fieldsToDelete: PropTypes.arrayOf(PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    index: PropTypes.number.isRequired,
+    name: PropTypes.string.isRequired,
+  })).isRequired,
+  onCancel: PropTypes.func.isRequired,
+  onConfirm: PropTypes.func.isRequired,
+  submitting: PropTypes.bool.isRequired,
+};
+
+const DeleteModal = ({
+  onConfirm,
+  onCancel,
+  fetchUsageStatistics,
+  fieldsToDelete,
+  submitting,
+}) => {
+  const [usageNumber, setUsageNumber] = useState(null);
+  const [usageNumberLoaded, setUsageNumberLoaded] = useState(false);
+
+  useEffect(() => {
+    const fetchStatistics = async () => {
+      const promises = fieldsToDelete.map(cf => {
+        return fetchUsageStatistics(cf.id);
+      });
+      const response = await Promise.all(promises);
+      const body = await Promise.all(response.map(res => res.json()));
+
+      const usageCount = body.reduce((acc, { fieldId, count }) => [
```

Comment:
Please name it `newUsageNumber` or something like this as there are two a bit different naming about the same thing `usageNumber` and `usageCount`.